test/contracts/HCElection.aes

```diff
@@ -125,7 +145,20 @@ main contract HCElection =
     pick_validator(n, (_, stake) :: validators)            = pick_validator(n - stake, validators)
 
   function assert_protocol_call() =
-      require(Call.caller == Contract.creator, "Must be called by the protocol")
+    require(Call.caller == Contract.creator, "Must be called by the protocol")
+
+  function calc_carry_over(carry_over_flag : bool) =
+    let ei = state.pin_reward
+    if (carry_over_flag)
+      ei.base + ei.carry_over
+    else
+      0
+
+  function calc_next_base_reward(next_base_pin_reward : int) =
```

Comment:
Yes, `-1` isn't super-elegant here...

Reply:
No, but people seem to be sensitive about adding extra complexity etc to the code, so now this parameter has two different meanings depending on it's value. I can't sem to figure out how to pass null or similar as a parameter (which is definitely a good thing that you don't).

Reply:
...or perhaps more specifically to the API, as this was actually complexity _added_ to the code.